x-pack/legacy/plugins/apm/server/lib/transaction_groups/fetcher.ts

```diff
@@ -32,38 +25,23 @@ export type Options = TopTransactionOptions | TopTraceOptions;
 
 export type ESResponse = PromiseReturnType<typeof transactionGroupsFetcher>;
 export function transactionGroupsFetcher(options: Options, setup: Setup) {
-  const { client, config, start, end, uiFiltersES } = setup;
+  const { client, config } = setup;
 
-  const bool: BoolQuery = {
-    must_not: [],
-    // prefer sampled transactions
-    should: [{ term: { [TRANSACTION_SAMPLED]: true } }],
-    filter: [
-      { range: rangeFilter(start, end) },
-      { term: { [PROCESSOR_EVENT]: 'transaction' } },
-      ...uiFiltersES
-    ]
-  };
+  const projection =
+    options.type === 'top_traces'
+      ? getTracesProjection({ setup })
+      : getTransactionGroupsProjection({
+          setup,
+          serviceName: options.serviceName,
+          transactionType: options.transactionType
+        });
 
-  if (options.type === 'top_traces') {
-    // A transaction without `parent.id` is considered a "root" transaction, i.e. a trace
-    bool.must_not.push({ exists: { field: PARENT_ID } });
-  } else {
-    bool.filter.push({ term: { [SERVICE_NAME]: options.serviceName } });
-    bool.filter.push({ term: { [TRANSACTION_TYPE]: options.transactionType } });
-  }
-
-  const params = {
-    index: config.get<string>('apm_oss.transactionIndices'),
+  const params = mergeProjection(projection, {
     body: {
       size: 0,
-      query: {
-        bool
-      },
       aggs: {
         transactions: {
           terms: {
-            field: TRANSACTION_NAME,
             order: { sum: 'desc' },
             size: config.get<number>('xpack.apm.ui.transactionGroupBucketSize')
           },
```

Comment:
One example where this `projection` abstraction readability issues is the comment further down:
```
// sort by _score to ensure that buckets with sampled:true ends up on top
```

How does the reader see whether `sampled:true` is added?

Reply:
Hmm, actually, I think the `should` clause should not be in the projection. The projection should only care about filtering, not about scoring. Do you think it would suffice if we move the `should` clause to this file?

Reply:
Sure, that would make it more clear if they are co-located (the sort and the `should`)